internal/statemachine/helper.go

```diff
@@ -619,29 +621,34 @@ func dpkgDivert(baseDir string, target string) (*exec.Cmd, *exec.Cmd) {
 	return execCommand("chroot", divert...), execCommand("chroot", undivert...)
 }
 
-// backupReplaceStartStopDaemon backup start-stop-daemon and replace it with a fake one
-// Returns a restore function to put the original one in place
-func backupReplaceStartStopDaemon(baseDir string) (func(error) error, error) {
-	const startStopDaemonContent = `#!/bin/sh
-echo
-echo "Warning: Fake start-stop-daemon called, doing nothing"
-`
+// divertExecWithFake replaces a target file in targetDir with a provided fake content
+// using dpkg-divert, and returns two commands lists: one for diverting, one for undiverting.
+func divertExecWithFake(targetDir string, file string, fakeContent string) ([]*exec.Cmd, []*exec.Cmd) {
+	divertCmd, undivertCmd := dpkgDivert(targetDir, file)
+
+	divertCmds := []*exec.Cmd{
+		divertCmd,
+		execCommand("chroot", targetDir, "sh", "-c", "echo '"+fakeContent+"' > "+file),
+		execCommand("chroot", targetDir, "chmod", "+x", file),
+	}
+	undivertCmds := []*exec.Cmd{
+		execCommand("chroot", targetDir, "rm", file),
+		undivertCmd,
+	}
 
-	startStopDaemon := filepath.Join(baseDir, "sbin", "start-stop-daemon")
-	return helper.BackupReplace(startStopDaemon, startStopDaemonContent)
+	return divertCmds, undivertCmds
 }
 
-// backupReplaceInitctl backup initctl and replace it with a fake one
-// Returns a restore function to put the original one in place
-func backupReplaceInitctl(baseDir string) (func(error) error, error) {
-	const initctlContent = `#!/bin/sh
-if [ "$1" = version ]; then exec /sbin/initctl.REAL "$@"; fi
-echo
-echo "Warning: Fake initctl called, doing nothing"
-`
+// divertStartStopDeamon diverts /usr/sbin/start-stop-daemon with one doing nothing.
+func DivertStartStopDaemon(targetDir string) ([]*exec.Cmd, []*exec.Cmd) {
```

Comment:
In relation to my previous comment: It feels wrong to have to export a function to then define an unexported variable in the same package, to finally enable monkey patching it for tests. I think that is a hint that we consider this piece of logic to be independent, and so making it obvious by moving it the helper package makes sense to me.